src/additional/phoneDK.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * Matches DK phone number format
+ *
+ * allows '-' or ' ' as a separator and allows parens around area code
+ * some people may want to put a '+45' in front of their number
+ *
+ * +45 21-77-21-77
+ * +45 21 77 21 77
+ * +45 21772177
+ * +4521772177
+ * 21 77 21 77
+ * 21-77-21-77
+ * 2177 2177
+ * 2177-2177
+ * 21772177
+ *
+ * but not
+ * +1 3243121212
+ * +45 112
+ * +452143284233
+ * 2143284233
+ */
+$.validator.addMethod( "phoneDK", function( phone_number, element ) {
+	phone_number = phone_number.replace( /\s+/g, "" );
+	return this.optional( element ) || phone_number.length > 9 &&
+		phone_number.match( /^(\+45( )?)?([0-9]{2})((-| )?[0-9]{2})((-| )?[0-9]{2})((-| )?[0-9]{2})$/ );
```

Comment:
Please use `regex.test(string)` instead of `string.match(regex)`. We only need to know whether the phone number matches the regex or not. We don't need to retrieve the matched parts.

Also, please use non-capturing instead of capturing groups [1]

[1] https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp#grouping-back-references

Reply:
okay thanks, will look on it later this week

Reply:
@Arkni can this regex codestyle be enforced via jshint?

Reply:
None that I am aware of.